src/pocketmine/network/mcpe/protocol/StructureBlockUpdatePacket.php

```diff
@@ -26,16 +26,32 @@
 #include <rules/DataPacket.h>
 
 use pocketmine\network\mcpe\NetworkSession;
+use pocketmine\network\mcpe\protocol\types\StructureEditorData;
 
-class StructureBlockUpdatePacket extends DataPacket{
+class StructureBlockUpdatePacket extends DataPacket/* implements ServerboundPacket*/{
 	public const NETWORK_ID = ProtocolInfo::STRUCTURE_BLOCK_UPDATE_PACKET;
 
+	/** @var int */
+	public $structureBlockX;
+	/** @var int */
+	public $structureBlockY;
+	/** @var int */
+	public $structureBlockZ;
+	/** @var StructureEditorData */
+	public $structureEditorData;
+	/** @var bool */
+	public $isStructureBlockPowered;
```

Comment:
These names don't need to be so verbose. It's already implied that they target a `structure block` because of the packet name.